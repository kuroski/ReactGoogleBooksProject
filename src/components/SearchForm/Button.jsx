import styled from 'styled-components'

const Button = styled.button`
    position: absolute;
    top: calc(50% - 12px);
    right: .5rem;
    background-color: transparent;
    border: none;
    outline: none;
    color: white;
    cursor: pointer;
`

export default Button