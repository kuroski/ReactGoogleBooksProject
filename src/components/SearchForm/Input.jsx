import styled from 'styled-components'

const Input = styled.input`
    background-color: #6D7993;
    padding: .5rem 1rem;
    flex: 1;
    border: none;
    outline: none;
    color: white;
    font-family: Roboto;
    font-size: 1.3rem;
    box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
`

export default Input