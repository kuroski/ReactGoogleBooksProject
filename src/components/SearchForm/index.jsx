import React, {Component, PropTypes} from 'react'
import styled from 'styled-components'
import Input from './Input'
import Button from './Button'

const Container = styled.form`
  display: flex;
  text-align: center;
  position: relative;
`;

const propTypes = {
  onSubmit: PropTypes.func.isRequired
}

class SearchForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      term: ''
    }
    this.setTerm = this.setTerm.bind(this)
    this.executeSearch = this.executeSearch.bind(this)
  }

  setTerm(event) {
    this.setState({term: event.target.value})
  }

  executeSearch(event) {
    event.preventDefault()
    this.props.onSubmit(this.state.term)
  }

  render() {
    return (
      <Container>
        <Input autoFocus value={this.state.term} onChange={this.setTerm} />
        <Button onClick={this.executeSearch}>
          <i className="material-icons">search</i>
        </Button>
      </Container>
    )
  }
}

SearchForm.propTypes = propTypes

export default SearchForm