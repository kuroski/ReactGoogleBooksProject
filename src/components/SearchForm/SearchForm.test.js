import React from 'react'
import SearchForm from './'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'

describe('SearchForm', () => {
  it('renders correctly', () => {
    const SearchFormTree = renderer.create(
      <SearchForm onSubmit={() => ''} />
    ).toJSON()
    expect(SearchFormTree).toMatchSnapshot()
  })

  it('should contain an input and a button', () => {
    const wrapper = shallow(<SearchForm onSubmit={() => ''} />)
    expect(wrapper.containsAllMatchingElements([
      <input />,
      <button>Add</button>
    ])).toEqual(true)
  })
})