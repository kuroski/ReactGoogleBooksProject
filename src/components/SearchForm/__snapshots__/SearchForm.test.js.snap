exports[`SearchForm renders correctly 1`] = `
<form
  className="ceDFYc">
  <input
    autoFocus={true}
    className="fCQilL"
    onChange={[Function]}
    placeholder="..."
    value="" />
  <button
    className="jyNRcu"
    onClick={[Function]}>
    <i
      className="material-icons">
      search
    </i>
  </button>
</form>
`;
